hal/src/nRF52840/ble_hal.cpp

```diff
@@ -741,6 +775,8 @@ os_thread_return_t BleObject::BleEventDispatcher::processBleEventFromThread(void
                 case BLE_GATTC_EVT_HVX: {
                     BleObject::getInstance().gattc()->processDataNotifiedEventFromThread(event);
                 }
```

Comment:
When getting a notification, it is calling both `processDataNotifiedEventFromThread(event)` and `processSecurityEventFromThread(event)`

```suggestion
                    break;
                }
```

Reply:
Good catch! 👍